api/src/main/java/org/eclipse/microprofile/jwt/builder/Jwt.java

```diff
@@ -0,0 +1,93 @@
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.builder;
+
+import java.util.Map;
+
+import org.eclipse.microprofile.jwt.builder.spi.JwtProvider;
+
+/**
+ * Factory class for creating {@link JwtClaimsBuilder} which produces
+ * signed, encrypted or signed first and then encrypted JWT tokens.
+ *
+ * <p>
+ * The following example shows how to initialize a {@link JwtClaimsBuilder} from an existing resource
+ * containing the claims in a JSON format and produce a signed JWT token with a configured signing key:
+ * 
+ * <pre>
+ * <code>
+ * String = Jwt.claims("/tokenClaims.json").sign();
+ * </code>
+ * </pre>
+ * <p>
+ * The next example shows how to use {@link JwtClaimsBuilder} to add the claims and encrypt a JSON
+ * representation of these claims with a configured encrypting key:
+ * 
+ * <pre>
+ * <code>
+ * String = Jwt.claims().issuer("https://issuer.org").claim("custom-claim", "custom-value").encrypt();
+ * </code>
+ * </pre>
+ * <p>
+ * The final example shows how to initialize a {@link JwtClaimsBuilder} from an existing resource
+ * containing the claims in a JSON format, produce an inner signed JWT token with a configured signing key
+ * and encrypt it with a configured encrypting key.
+ * 
+ * <pre>
+ * <code>
+ * String = Jwt.claims("/tokenClaims.json").innerSign().encrypt();
+ * </code>
+ * </pre>
+ */
+public final class Jwt {
+    private Jwt() {
+        
+    }
+    /**
+     * Creates a new instance of {@link JwtClaimsBuilder}
+     *
+     * @return {@link JwtClaimsBuilder}
+     */
+    public static JwtClaimsBuilder claims() {
+        return JwtProvider.provider().claims();
+    }
+
+    /**
+     * Creates a new instance of {@link JwtClaimsBuilder} from a map of claims.
+     * 
+     * @param claims the map with the claim name and value pairs. Claim value is converted to String unless it is
+     *        an instance of {@code Boolean}, {@code Number}, {@code Collection}, {@code Map},
+     *        {@code JsonObject} or {@code JsonArray}.
+     * @return {@link JwtClaimsBuilder}
+     */
+    public static JwtClaimsBuilder claims(Map<String, Object> claims) {
+        return JwtProvider.provider().claims(claims);
+    }
+
+    /**
+     * Creates a new instance of {@link JwtClaimsBuilder} from a JSON resource.
+     * 
+     * @param jsonLocation JSON resource location
+     * @return {@link JwtClaimsBuilder}
+     */
+    public static JwtClaimsBuilder claims(String jsonLocation) {
```

Comment:
Support for URI so that resource within archive can be used? (within a jar contained within the WAR, for example)

Reply:
Or should we just update the docs to note it can be a location pointing to a class path, filesystem or the HTTP location. If we introduce `URI` then `String` would need to be restricted as to what it may contain and I'd rather follow the the existing MP JWT guideline (for the key location, etc) ?

Reply:
Yes, I wanted to change parameter type from String to URI.

With String, anything can be put into it but we interpret it as a File path location.